src/sdltiles.cpp

```diff
@@ -480,6 +527,11 @@ void WinDestroy()
 #endif
     clear_texture_pool();
 
+    if( alternate_rect_tex ){
+        SDL_DestroyTexture( alternate_rect_tex );
```

Comment:
Here you could unconditionally (and without checking `alternate_rect_tex`) call `alternate_rect_tex.reset();`.